rosbag2_performance/writer_benchmarking/src/main.cpp

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020, Robotec.ai sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include "writer_benchmark.hpp"
+
+// TODO(adamdbrw) the benchmkark being ROS node is not necessary, only used for logging
+// and parameters. Otherwise ROS dependencies are calls to rcl and rmw.
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+  rclcpp::Context ctx;
```

Comment:
```suggestion
```

Reply:
I would suspect you'd need a `rclcpp::spin(bench)` to actually make the node react to the parameters.

Reply:
The node apparently reacts to the parameters given by command line without the need to spin